### ISSUE TITLE
include foldable regions in outline

### ISSUE THREAD

**Issue by esotericist** [maptz/maptz.vscode.extensions.customfolding#90]
i'm doing a lot of work in c, and i was wanting a better way to do grouping of logical units within a file. this extension is a good help, but i was a little surprised that the regions don't show up in the outline. is it possible to do that?

**Comment by maptz** [comment]
I'm not sure. I'll have to look into it. I'm sure it must be possible to interact with the outliner.

Do you have an example of how you think it should appear?

**Comment by esotericist** [comment]
i'm very sorry for never following up on this. basically what i'd like is to simply insert the name of regions in between functions in the outline. for instance i have in one file:
![image](https://user-images.githubusercontent.com/1569754/159185763-eeae35e9-90be-44ab-9b2c-1d79a1b0e06d.png)

and i'd like 'event prims' to somehow show up in between `events_teardown() `and `prim(event_timer)` in the outline here
![image](https://user-images.githubusercontent.com/1569754/159185793-60718ded-b6cf-422f-9874-993320b084f9.png)

bonus points if i can click on it to go to the foldable region, but that's not actually a requirement.

basically i just want it to function as a visible divider between groups of logical units in the outline. (in this case, i am using foldable regions to group primitives for a virtual machine)

**Comment by maptz** [comment]
I believe this might be possible. 


According to [this](https://stackoverflow.com/questions/48306520/how-can-i-implement-my-own-code-outline-layout-in-vscode) link it should be possible to add custom elements to the outline. I'll try to look into it a bit more.

**Comment by maptz** [comment]
Hi, Due to time constraints, I'm not going to prioritise this enhancement for the moment on the extension. If you're able to help, then please let me know.

**Comment by maptz** [comment]
I've added this as an experimental feature in version 1.0.20. It's behind an option flag, so you have to turn it on. 

Add the following to your `settings.json` file:

```json
{
   "maptz.regionfolder.showRegionsInOutline": true
}
```
